koap/src/commonMain/kotlin/Message.kt

```diff
@@ -389,11 +395,11 @@ sealed class Message {
         override val payload: ByteArray
     ) : Message() {
 
-        enum class Type {
-            Confirmable,
-            NonConfirmable,
-            Acknowledgement,
-            Reset,
+        sealed class Type() {
```

Comment:
```suggestion
        sealed class Type {
```

Reply:
Surprised the linter didn't pick up on this one.

Reply:
I think the standard set of lint rules is pretty conservative, i.e. there is a much larger set of "experimental" lint rules we can enable (maybe we should), and I'd hope a lint rule is in there that would catch this. 🤷